app/models/ems_refresh/save_inventory.rb

```diff
@@ -246,12 +246,39 @@ def save_networks_inventory(hardware, hashes, mode = :refresh)
   def save_system_services_inventory(parent, hashes, mode = :refresh)
     return if hashes.nil?
 
-    deletes = case mode
-              when :refresh then nil
-              when :scan    then :use_association
-              end
+    #######
+    # tripleo specific
+    #######
+
+    # if parent is OpenStack Cloud
+    # and if parent have Infra provider
+    if parent.kind_of?(ManageIQ::Providers::Openstack::CloudManager)
```

Comment:
For the pluggable providers, we are trying not to add any more specific providers into the generic code. So there should not be a condition about Openstack, can it be rewritten without that?